packages/eth2.0-config/src/index.ts

```diff
@@ -1,6 +1,7 @@
-import {IBeaconParams} from "../params";
+import {IBeaconParams} from "@chainsafe/eth2.0-params";
+import {createIBeaconSSZTypes} from "@chainsafe/eth2.0-ssz-types";
+
 import {IBeaconConfig} from "./interface";
-import {createIBeaconSSZTypes} from "../sszTypes";
 
 export * from "./interface";
```

Comment:
could you export mainet and minimal config here. Otherwise it will be weird import `import {config} from "@chainsafe/eth2.0-config/lib/presets/mainnet"`

Reply:
The reason I didn't import/export to index is that I didn't want to add the overhead of always creating preset configurations when they aren't needed. When we export the presets to the root, we will be wrongfully initializing all of these singletons.

I was viewing presets as an "extra", even though its practically invaluable. The main export is the interface and the function for a DIY configuration. For "convenience", we have the well-known instantiations as singletons.

I would advocate leaving as is or splitting out presets into separate packages.

Reply:
Generated presets are like what 2-3kb at most? practically irrelevant! But for convention and for IDE support (non root exports are not suggested out-of-box) it's quite important to export from index.

Reply:
I agree its somewhat small, but I think its bad to assume that its reasonable in all cases to be doing this extra work. If in the future we expand the number of presets, then this wasted cpu/memory just grows.

I think a good path forward is making `eth2.0-presets-mainnet` and `eth2.0-presets-minimal` packages that export the relevant singletons (an IBeaconParams, IBeaconSSZTypes, and IBeaconConfig)

Reply:
Dunno, I think even if it grows to 1k objects it will still be irrelevant to cpu/memory time. Last time I checked,  generation of types was less than 0.0001% on stack. I think you are nit picky here :stuck_out_tongue: , we can safely export it for developer convenience,  if it shows any relevant cpu/memory impact I'm up for splitting into separate packages

Reply:
we decided to leave as is for now